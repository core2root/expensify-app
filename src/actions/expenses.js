import uuid from 'uuid';

//Add
export const addExpense = (
  { description = '', note = '', amount = 0, createdAt = 0 }
    = {}) => ({
      type: 'ADD_EXPENSE',
      expense: {
        id: uuid(),
        description: description,
        note: note,
        amount: amount,
        createdAt: createdAt
      }
    })

//Remove
export const removeExpense = ({ id } = {}) => ({
  type: 'REMOVE_EXPENSE',
  id: id
});

//Edit
export const editExpense = (id, updates) => ({
  type: 'EDIT_EXPENSE',
  id: id,
  updates: updates
});