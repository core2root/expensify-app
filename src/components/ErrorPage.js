import React from 'react';
import {Link} from 'react-router-dom';

const ErrorPage =()=>(
  <div>
    <p>Oops! looks like this page doesen't exist <b>CODE: 404</b> </p>
    <Link to='/'>Go Home</Link>
  </div>
);

export default ErrorPage;