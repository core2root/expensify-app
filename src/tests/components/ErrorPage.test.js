import {shallow} from 'enzyme';
import React from 'react';
import expenses from '../fixtures/expenses';
import ErrorPage from '../../components/ErrorPage';

test('should render page not found correctly', () => {
  const wrapper = shallow(<ErrorPage />);
  expect(wrapper).toMatchSnapshot();
});
