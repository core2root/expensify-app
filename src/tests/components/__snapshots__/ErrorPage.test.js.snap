// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render page not found correctly 1`] = `
<div>
  <p>
    Oops! looks like this page doesen't exist 
    <b>
      CODE: 404
    </b>
     
  </p>
  <Link
    replace={false}
    to="/"
  >
    Go Home
  </Link>
</div>
`;
